### ISSUE TITLE
Figma Component Upgrades

### ISSUE THREAD

**Issue by SOjaHPE** [grommet/hpe-design-system#2450]
[Main exploration file
](https://www.figma.com/file/IpxvTNHkD0Oz2GXw3KWgcw/Audit---HPE-Design-System-Library?node-id=0%3A1)

Future Sprints are being tracked in the [Figma 2.0 Project board](https://github.com/orgs/grommet/projects?type=beta)

## Goals
The DS component set in Figma requires a Spring Cleaning initiative to:
1) Connect with internal design systems to provide enhancements to support their BU use cases
2) Address all Figma componentry related bugs (auto-layout issues, responsiveness, text wrapping, etc.)
3) Create clear guidelines for how to create new components as well as how to deprecate old components
4) Create and deprecate components as needed
5) **PROVIDE GLCP WITH BUG FIXES AND NEW COMPONENTS FOR THEIR PAL RELEASE AT THE END OF APRIL**

---
#### Deprecating old components
- [x] #2483
- [x] #2511 

#### Creating new components
- [x] #2473
- [x] #2495
- [x] Explore interactive componentry and how they can fit within this audit
- [x] Get feedback from stakeholders
- [x] Explore plugins or possibly create plugins that can help us with the Audit
- [x] #2235 

#### Release planning
- [x] Strategize on how to release new components while deprecating old ones
- [x] Explore ways and avenues to communicate audit changes
- [x] Create guidelines/instructions for designers to best assimilate with new changes

#### Prioritizing components
- [x] [List of all broken components in the DS](https://www.figma.com/file/IpxvTNHkD0Oz2GXw3KWgcw/Audit---HPE-Design-System-Library?node-id=0%3A1)
- [x] Connect with stakeholders for a list of broken components as well as a list of overlapping components from their design system and ours (check resources section)
- [x] #2482
- [x] Setup the new DS 2.0 component file
- [x] Fix bugs, create components, and deprecate components based on prioritization
- [ ] Create a list/chart/graphic of all deprecated and new components. Where will this live?

---
### Initial Release Checklist
- [x] #2711 
- [x] #2694 
- [x] #2722
- [x] Figure out what to do with old component files
- [x] Move over components to new file
- [x] Include update log for each component page and an update log for major updates
- [x] Add finishing touches to the new file

---

#### Ezmeral
- Point person: Ian Erickson, Geetha Priya Padmanaban
- [Design System](https://www.figma.com/file/yZyMsoTk6zSn2TgHH257Ea/Ezmeral-Components-V2?node-id=0%3A1)
- [Bug report tracker](https://www.figma.com/file/yZyMsoTk6zSn2TgHH257Ea/Ezmeral-Components-V2?node-id=2133%3A64065)
- [Chart of overlapping components](https://www.figma.com/file/UHoXXKufEnlPQkMpcQOld4/UI-Design-Debt-(GLCP)?node-id=5%3A2086)

#### GLCP
- Point person: Sylvia Shen
- [PAL Design System](https://www.figma.com/file/6hYD74saV26sSgoLd3Fu9q/PAL-(Platform-Asset-Library)?node-id=0%3A1)
- [Bug report tracker](https://www.figma.com/file/UHoXXKufEnlPQkMpcQOld4/UI-Design-Debt-(GLCP)?node-id=5%3A2091)
- [Chart of overlapping components](https://www.figma.com/file/UHoXXKufEnlPQkMpcQOld4/UI-Design-Debt-(GLCP)?node-id=5%3A2086)

#### Storage
- Point person: Sherry Mead
- Design System (See Pattern review, library, drafts)
- [Bug report tracker](https://www.figma.com/file/IpxvTNHkD0Oz2GXw3KWgcw/Audit---HPE-Design-System-Library?node-id=471%3A9575) 
- [Chart of overlapping components](https://www.figma.com/file/UHoXXKufEnlPQkMpcQOld4/UI-Design-Debt-(GLCP)?node-id=5%3A2086)

**Comment by KennyAtHPE** [comment]
Main findings so far:
- The different design systems have created a variety of templates for quick use (forms, lists, accordions, etc.). Unfortunately, they are also creating new components for the building blocks of these templates as opposed to using the Design System's component
- There are also some use cases where they are renaming the new components and not aligning themselves with the Design System. This will cause issues for people searching for assets
- Our older components are missing auto layouts alongside other bugs
- Most of our instances are disorganized

**Comment by KennyAtHPE** [comment]
To Do list moving forward:

- [ ] Create a list of all broken components in the DS
- [ ] Connect with Sylvia for a list of broken components
- [ ] Connect with Ezmeral and Storage on the audit and solicit help on any broken components they may have noticed
- [ ] Cross-reference our list and the other platforms to prioritize component fixes

**Comment by bahriaditi** [comment]
@KennyAtHPE @SOjaHPE and I just did a quick sync on the DS 2.0 work. Here's what we discussed at next steps:

- Begin tackling issue #2482 
- In terms of priorities - focus on PAL components needs _(Kenny to reach out to Sylvia)_
- #2495 will be ongoing and continue to be worked on in parallel

- [ ] Aditi will set up a work session to tackle #2235 in the next couple of weeks

- [ ] Aditi to set up recurring weekly design syncs on DS 2.0

**Comment by ericsoderberghp** [comment]
Will track future work on a per-component basis
